app/services/member_with_authentication.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class MemberWithAuthentication
+  include ActiveModel::Model
```

Comment:
How does this class work? It pretends to be an ActiveRecord model for the validation? A header comment would be great on this one.